### ISSUE TITLE
ubuntu 14.10 cmake pthread"s"

### ISSUE THREAD

**Issue by anselm** [OSVR/libfunctionality#11]
There is a documented issue with cmake and pthreads that seems to be being triggered here. Haven't had a chance to look at it more closely... Here is how to reproduce it under Ubuntu 14.10:

> cd libfunctionality
> mkdir build
> cd build
> cmake ..
> cat CMakeFiles/CMakeError.log

CheckSymbolExists.c:(.text+0x16): undefined reference to `pthread_create'
collect2: error: ld returned 1 exit status
make[1]: *** [cmTryCompileExec715743377] Error 1
make[1]: Leaving directory `/root/libfunctionality/build/CMakeFiles/CMakeTmp'
make: *** [cmTryCompileExec715743377/fast] Error 2

File /root/libfunctionality/build/CMakeFiles/CMakeTmp/CheckSymbolExists.c:
/* */
#include <pthread.h>

int main(int argc, char** argv)
{
  (void)argv;
#ifndef pthread_create
  return ((int*)(&pthread_create))[argc];
#else
  (void)argc;
  return 0;
#endif
}

**Comment by godbyk** [comment]
The errors in that file do not indicate project build errors. Those errors are the result of CMake testing various pthread implementations (namely, various `pthread_create` possibilities). You'll notice in the CMake output that it says something like:
```
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
```

The first two `not found` notices correspond to the errors you see in that log file. The third attempt succeeds. CMake will use that final method when building the code.

In short, this is the expected behavior and not a bug.
